sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/CanaryStrategy.java

```diff
@@ -85,6 +85,11 @@ public CanaryStrategy(Strategy<Step> postCanaryStrategy, int requiredProceeds, L
     }
 
     @Override
+    public String getName() {
+        return strategy.toString() + "-canary";
```

Comment:
toString -> getName

Reply:
FWIW should probably get this when feasible, it'll currently likely result in something like `<DEFAULT JAVA OBJECT DUMP>-canary`